File: src/main/java/org/funivan/lologa/tile/Position/Position.java
package org.funivan.lologa.tile.Position;

public class Position implements PositionInterface {
    private final int row;
    private final int col;

    public Position(int row, int col) {
        this.row = row;
        this.col = col;
    }

    @Override
    public int row() {
        return this.row;
    }

    @Override
    public int col() {
        return this.col;
    }

    @Override
    public String toString() {
        return "Position{" + this.row + "x" + this.col + '}';
    }

    @Override
    public boolean equals(Object o) {
        boolean result = false;
        if (this == o) {
            result = true;
        } else if (o instanceof PositionInterface) {
            result = (((PositionInterface) o).row() == this.row() && ((PositionInterface) o).col() == this.col());
        }
        return result;
    }

    @Override
    public int hashCode() {
        return 31 * this.row + this.col;
    }

}

File: src/main/java/org/funivan/lologa/tile/Position/Position.kt
package org.funivan.lologa.tile.Position

class Position(private val row: Int, private val col: Int) : PositionInterface {

    override fun row(): Int {
        return this.row
    }

    override fun col(): Int {
        return this.col
    }

    override fun toString(): String {
        return "Position{" + this.row + "x" + this.col + '}'
    }

    override fun equals(o: Any?): Boolean {
        var result = false
        if (this === o) {
            result = true
        } else if (o is PositionInterface) {
            result = o.row() == this.row() && o.col() == this.col()
        }
        return result
    }

    override fun hashCode(): Int {
        return 31 * this.row + this.col
    }

}
